packages/binding-coap/src/coap-server.ts

```diff
@@ -136,81 +137,145 @@ export default class CoapServer implements ProtocolServer {
         }
     }
 
-    public expose(thing: ExposedThing, tdTemplate?: WoT.ExposedThingInit): Promise<void> {
-        let urlPath = slugify(thing.title, { lower: true });
+    private createThingUrlPath(thing: ExposedThing) {
+        const urlPath = slugify(thing.title, { lower: true });
 
         if (this.things.has(urlPath)) {
-            urlPath = Helpers.generateUniqueName(urlPath);
+            return Helpers.generateUniqueName(urlPath);
         }
+
+        return urlPath;
+    }
+
+    private createCoreResource(urlPath: string): void {
         this.coreResources.set(urlPath, { urlPath, parameters: thingDescriptionParameters });
+    }
+
+    // TODO: Could probably be defined as a general helper function
```

Comment:
That sounds like a good idea 👍

Reply:
@JKRhb do you still plan to create it as a general helper function? Maybe even in `core`?